Sources/TensorFlow/Core/Threading.swift

```diff
@@ -22,43 +22,83 @@ import Glibc
 
 // A thread that runs the provided body.
 class Thread {
-    var thread: pthread_t
-    init(perform body: @escaping () -> ()) {
-        class ThreadArg {
-        var body : () -> ()
-            init(body : @escaping () -> ()) {
-                self.body = body
-            }
-        }
-#if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
-        typealias ThreadBody = @convention(c) (UnsafeMutableRawPointer) -> UnsafeMutableRawPointer?
+#if os(Windows)
+    typealias Handle = HANDLE
+#else
+#if os(macOS) || os(iOS) || os(tvOS) || os(watchOS)
+    typealias Handle = pthread_t?
+#else
+    typealias Handle = pthread_t
+#endif
+#endif
+
+    static func _invalid_handle() -> Handle {
+#if os(Windows)
+      return INVALID_HANDLE_VALUE
 #else
-        typealias ThreadBody = @convention(c) (UnsafeMutableRawPointer?) -> UnsafeMutableRawPointer?
+#if os(macOS) || os(iOS) || os(tvOS) || os(watchOS)
+      return nil
+#else
+      return pthread_t()
+#endif
 #endif
-        let threadBody: ThreadBody = { arg in
+    }
+
+    class Procedure {
```

Comment:
@compnerd: why is a `Procedure` abstraction necessary? Is `typealias Procedure = () -> ()` sufficient instead?